Source/Private/Source.cpp

```diff
@@ -71,10 +93,76 @@ bool CheckMap(BYTE* fileBuffer, int bufferLen, noeRAPI_t* rapi)
 	return 1;
 }
 
+template<bool bBigEndian>
+bool CheckDZArchive(BYTE* fileBuffer, int bufferLen, noeRAPI_t* rapi)
+{
+	uint8_t ed = *(uint8_t*)(fileBuffer);
```

Comment:
```suggestion
	uint8_t ed = fileBuffer[0];
```
A byte is already a byte.

CheckMap above has a bufferLen check, but this one does not. For robustness (since there is no guarantee in the contract that bufferLen > 0):

```suggestion
	uint8_t ed = (bufferLen > 0) ? fileBuffer[0] : 0;
```

Otherwise may be able to crash this library by just opening an empty file.